test/functional/unattended_controller_test.rb

```diff
@@ -358,6 +358,47 @@ class UnattendedControllerTest < ActionController::TestCase
   end
 
   test 'host template provision URL can be generated from routes' do
-    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision'}
+    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision', :format => 'text'}
+  end
+
+  test "should not render a template to anonymous user" do
+    User.current = nil
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}
+    assert_response :redirect
+  end
+
+  test "should not render a template to user w/o email" do
+    User.current = FactoryGirl.create(:user)
+    User.current.auth_source.stubs(:authenticated?).returns(false)
+    session = SETTINGS[:login] ? {:user => User.current.id, :expires_at => 5.minutes.from_now} : {}
```

Comment:
I am not sure if I want to use fixtures, it's a step back actually. Time to provide `set_session_user` for FG objects?